exasol_integration_test_docker_environment/lib/test_environment/ports.py

```diff
@@ -0,0 +1,40 @@
+import socket
+from contextlib import ExitStack
+from typing import List, Optional
+
+
+def find_free_ports(num_ports: int) -> List[int]:
+    def new_socket():
+        return socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    def bind(sock: socket.socket, port: int):
+        sock.bind(('', port))
+        sock.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+
+    result = list()
+    with ExitStack() as stack:
+        sockets = [stack.enter_context(new_socket()) for dummy in range(num_ports)]
+        for sock in sockets:
+            bind(sock, 0)
+            result.append(sock.getsockname()[1])
+    with ExitStack() as stack:
+        sockets_and_ports = [(stack.enter_context(new_socket()), port) for port in result]
+        for sock, port in sockets_and_ports:
+            bind(sock, port)
```

Comment:
extract into function check_port_numbers

Reply:
this is more or less a double check pattern, if I see this correct

Reply:
I must admit I didn't understand the implementation strategy completely, probably my knowledge about sockets and ports is incomplete.

What I did is just 
* move the logic from `testing/utils.py` to the new file `ports.py` which is more specific and intuitive in my eyes. 
* apply some additional refactoring to reduce code duplication

Reply:
code will be extract into function `check_port_numbers()` in next push